indico/modules/rb_new/operations/conflicts.py

```diff
@@ -57,13 +57,16 @@ def get_rooms_conflicts(rooms, start_dt, end_dt, repeat_frequency, repeat_interv
     for room_id, occurrences in blocked_rooms.iteritems():
         rooms_conflicts[room_id] += get_room_blockings_conflicts(room_id, candidates, occurrences)
 
-    # TODO: do proper per-room override checks
-    if not rb_is_admin(session.user):
+    if not (allow_admin and rb_is_admin(session.user)):
         for room_id, occurrences in nonbookable_periods.iteritems():
-            rooms_conflicts[room_id] += get_room_nonbookable_periods_conflicts(candidates, occurrences)
+            room = Room.get_one(room_id)
```

Comment:
how man rooms do we usually have here? and were they queried before (in that case sqlalchemy has them in its identity cache)? if not, this triggers one query per room!